src/components/SettingsDialog/Components/SettingsInput.tsx

```diff
@@ -0,0 +1,31 @@
+import {ChangeEvent, FC} from "react";
+import "./SettingsInput.scss";
+
+export interface SettingsInputProps {
+  label: string;
+  value: string;
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  submit: () => void;
+  disabled?: boolean;
+  type?: "text" | "password";
+  placeholder?: string;
+}
+
+export const SettingsInput: FC<SettingsInputProps> = ({label, value, onChange, submit, disabled, type, placeholder, children}) => (
+  <div className="settings-input__container">
+    <input
+      className={!placeholder ? "settings-input__hidden-placeholder" : undefined}
+      placeholder={placeholder ?? label}
+      value={value}
+      onChange={onChange}
+      onBlur={() => value && submit()}
+      onKeyDown={(e) => e.key === "Enter" && value && submit()}
+      disabled={disabled}
+      type={type ?? "text"}
+      id={label}
```

Comment:
Mmmhhh... I'd rather pass a separate id here as a prop. E.g. if we'll have the same label for different inputs like the ~~board~~ name and ~~user~~ name in some language file.